chain-crypto/src/algorithms/vrf/dleq.rs

```diff
@@ -11,26 +9,26 @@ pub struct Proof {
     z: Scalar,
 }
 
-const PROOF_SIZE: usize = 64; // Scalar is 32 bytes
-
 impl Proof {
+    pub const PROOF_SIZE: usize = Scalar::BYTES_LEN * 2;
+
     pub fn to_bytes(&self, output: &mut [u8]) {
-        assert_eq!(output.len(), PROOF_SIZE);
-        output[0..32].copy_from_slice(self.c.0.as_bytes());
-        output[32..64].copy_from_slice(self.z.as_bytes());
+        assert_eq!(output.len(), Self::PROOF_SIZE);
+        output[0..32].copy_from_slice(&self.c.0.to_bytes());
```

Comment:
I would rather use `Scalar::BYTES_LEN` instead of hardcoding `32`. If we change the length of the non-default backend such error would be caught very late.
Same thing applies to wherever we use such constants.

Reply:
great, I've added tests to ensure these would have been cached